src/ui/hexedit.cc

```diff
@@ -519,20 +595,165 @@ void HexEdit::contextMenuEvent(QContextMenuEvent *event) {
   menu_.exec(event->globalPos());
 }
 
+void HexEdit::processMoveEvent(QKeyEvent *event) {
+  if (event->matches(QKeySequence::MoveToNextChar)) {
+    setSelection(curentPosition_ + 1, selectionSize_);
+    if (!isByteVisible(curentPosition_)) {
+      scrollRows(1);
+    }
+  }
+
+  if (event->matches(QKeySequence::MoveToPreviousChar)) {
+    setSelection(curentPosition_ - 1, selectionSize_);
+    if (!isByteVisible(curentPosition_)) {
+      scrollRows(-1);
+    }
+  }
+
+  if (event->matches(QKeySequence::MoveToNextLine)) {
+    setSelection(curentPosition_ + bytesPerRow_, selectionSize_);
+    if (!isByteVisible(curentPosition_)) {
+      scrollRows(1);
+    }
+  }
+
+  if (event->matches(QKeySequence::MoveToPreviousLine)) {
+    setSelection(curentPosition_ - bytesPerRow_, selectionSize_);
+    if (!isByteVisible(curentPosition_)) {
+      scrollRows(-1);
+    }
+  }
+
+  if (event->matches(QKeySequence::MoveToNextPage)) {
+    setSelection(curentPosition_ + bytesPerRow_ * rowsOnScreen_, selectionSize_);
+    scrollRows(rowsOnScreen_);
+  }
+
+  if (event->matches(QKeySequence::MoveToPreviousPage)) {
+    setSelection(curentPosition_ - bytesPerRow_ * rowsOnScreen_, selectionSize_);
+    scrollRows(-rowsOnScreen_);
+  }
+
+  if (event->matches(QKeySequence::MoveToStartOfLine)) {
+    setSelection(curentPosition_ - curentPosition_ % bytesPerRow_, selectionSize_);
+  }
+
+  if (event->matches(QKeySequence::MoveToEndOfLine)) {
+    setSelection(curentPosition_ + bytesPerRow_ - (curentPosition_ % bytesPerRow_) - 1, selectionSize_);
+  }
+
+  if (event->matches(QKeySequence::MoveToStartOfDocument)) {
+    setSelection(0, selectionSize_, true);
```

Comment:
Could you also comment bool arguments? It's hard to say what's this "true" means here. In this case something like this would be nice:
```C++
setSelection(0, selectionSize_, /*set_visible=*/true);
```